lib/success.js

```diff
@@ -15,43 +15,62 @@ export default async (pluginConfig, context) => {
     commits,
     releases,
   } = context;
-  const { gitlabToken, gitlabUrl, gitlabApiUrl, successComment, proxy } = resolveConfig(pluginConfig, context);
+  const { gitlabToken, gitlabUrl, gitlabApiUrl, successComment, successCommentCondition, proxy } = resolveConfig(
+    pluginConfig,
+    context
+  );
   const repoId = getRepoId(context, gitlabUrl, repositoryUrl);
   const encodedRepoId = encodeURIComponent(repoId);
   const apiOptions = { headers: { "PRIVATE-TOKEN": gitlabToken } };
 
   if (successComment === false) {
     logger.log("Skip commenting on issues and pull requests.");
+    logger.error(`Disabling commenting on issues and pull requests should be done using 'successCommentCondition'.
+Using 'false' for 'successComment' is deprecated and will be removed in a future major version!`);
   } else {
     const releaseInfos = releases.filter((release) => Boolean(release.name));
     try {
       const postCommentToIssue = (issue) => {
-        const issueNotesEndpoint = urlJoin(gitlabApiUrl, `/projects/${issue.project_id}/issues/${issue.iid}/notes`);
-        debug("Posting issue note to %s", issueNotesEndpoint);
-        const body = successComment
-          ? template(successComment)({ ...context, issue, mergeRequest: false })
-          : getSuccessComment(issue, releaseInfos, nextRelease);
-        return got.post(issueNotesEndpoint, {
-          ...apiOptions,
-          ...proxy,
-          json: { body },
-        });
+        const canCommentOnIssue = successCommentCondition
```

Comment:
With this setup we will always have some additional API calls even if we do not want to comment on an issue. Setting `successComment` to `false` had the advantage of an early stop. This is - currently - not possible and I have no idea if it is possible with `successCommentCondition`.